proof/access-control/Access.thy

```diff
@@ -565,202 +197,63 @@ where
-                        \<in> state_asids_to_policy_aux aag caps asid_tab vrefs"
-  | sata_asid_lookup: "\<lbrakk> asid_tab (asid_high_bits_of asid) = Some poolptr;
-                           (pdptr, VSRef (asid && mask asid_low_bits)
-                                         (Some AASIDPool), a) \<in> vrefs poolptr \<rbrakk>
-                       \<Longrightarrow> (pasASIDAbs aag asid, a, pasObjectAbs aag pdptr)
-                              \<in> state_asids_to_policy_aux aag caps asid_tab vrefs"
-  | sata_asidpool:    "\<lbrakk> asid_tab (asid_high_bits_of asid) = Some poolptr; asid \<noteq> 0 \<rbrakk> \<Longrightarrow>
-                        (pasObjectAbs aag poolptr, ASIDPoolMapsASID, pasASIDAbs aag asid)
-                                             \<in> state_asids_to_policy_aux aag caps asid_tab vrefs"
-
-fun cap_irqs_controlled :: "cap \<Rightarrow> irq set"
-where
+fun cap_irqs_controlled :: "cap \<Rightarrow> irq set" where
   "cap_irqs_controlled cap.IRQControlCap = UNIV"
-  | "cap_irqs_controlled (cap.IRQHandlerCap irq) = {irq}"
-  | "cap_irqs_controlled _ = {}"
+| "cap_irqs_controlled (cap.IRQHandlerCap irq) = {irq}"
+| "cap_irqs_controlled _ = {}"
 
-inductive_set state_irqs_to_policy_aux for aag caps
-where
-  sita_controlled: "\<lbrakk> caps ptr = Some cap; irq \<in> cap_irqs_controlled cap \<rbrakk>
-  \<Longrightarrow> (pasObjectAbs aag (fst ptr), Control, pasIRQAbs aag irq) \<in> state_irqs_to_policy_aux aag caps"
+inductive_set state_irqs_to_policy_aux for aag caps where
+  sita_controlled:
+    "\<lbrakk> caps ptr = Some cap; irq \<in> cap_irqs_controlled cap \<rbrakk>
+       \<Longrightarrow> (pasObjectAbs aag (fst ptr), Control, pasIRQAbs aag irq) \<in> state_irqs_to_policy_aux aag caps"
 
-abbreviation state_irqs_to_policy
-where
+abbreviation state_irqs_to_policy where
   "state_irqs_to_policy aag s \<equiv> state_irqs_to_policy_aux aag (caps_of_state s)"
 
-definition irq_map_wellformed_aux
-where
+definition irq_map_wellformed_aux where
   "irq_map_wellformed_aux aag irqs \<equiv> \<forall>irq. pasObjectAbs aag (irqs irq) = pasIRQAbs aag irq"
 
-abbreviation irq_map_wellformed
-where
+abbreviation irq_map_wellformed where
   "irq_map_wellformed aag s \<equiv> irq_map_wellformed_aux aag (interrupt_irq_node s)"
 
-definition state_vrefs
-where
-  "state_vrefs s = case_option {} vs_refs_no_global_pts o kheap s"
-
-abbreviation state_asids_to_policy
-where
-  "state_asids_to_policy aag s \<equiv> state_asids_to_policy_aux aag (caps_of_state s)
-        (arm_asid_table (arch_state s)) (state_vrefs s)"
-
-definition tcb_states_of_state
-where
-  "tcb_states_of_state s \<equiv> \<lambda>p. option_map tcb_state (get_tcb p s)"
-
-(* FIXME: RENAME: should be named something thread_st_auth. thread_states is confusing as
-   it outputs autorities *)
-definition thread_states
-where
+(* FIXME: rename to thread_st_auth or similar since it outputs authorities *)
```

Comment:
We should settle on a FIXME tag so we can find these again. Maybe`FIXME ACSPLIT`? (up to you, if you find something shorter that makes sense, all good)